lockfiles/README.md

```diff
@@ -0,0 +1,14 @@
+## Description
+
+The goal of these files is to be able to have a reproductible build of `irmin`
+for a benchmark purpose. It is used with `opam monorepo` and `dune`.
+
+## Disclamer
+
+Files older than september 21 only work in the context of `irmin-bench`. They
+are intended to be used with the following configuration:
+  - `ocaml.0.13.x`
+  - `opam-monorepo.0.2.7`
+  - `dune.2.9.3`
+
+:warning: We can't predict the behaviour if it is used with another configuration!
```

Comment:
I'm not totally sure to understand this warning. Why don't you set these versions in the lock files?

Reply:
There are two issues:
 * `opam monorepo` is needed to make `opam monorepo` works of course. As we need `dune` to build the code and `ocaml` to run it, even if we set the version in the lock file, it won't install it (or we need to use it with  `opam install [..] --locked` but it doesn't take advantage of `monorepo`
 * Some errors are raised when you are trying to compile `irmin` (monorepo version) with `dune.3.0.x` as it removes the builtin of `uchar`. If we want it to be compatible it requires to update the code for each branch. As we wanted to have the benchmarks with lock files quickly, we have decided to just pin the `dune` dependency.

Does it seem ok for you? :/